engine/src/main/scala/cromwell/engine/workflow/lifecycle/WorkflowFinalizationActor.scala

```diff
@@ -50,26 +50,30 @@ case class WorkflowFinalizationActor(workflowId: WorkflowId, workflowDescriptor:
   val tag = self.path.name
   val backendAssignments = workflowDescriptor.backendAssignments
 
-  override val successState = WorkflowFinalizationSucceededState
+  override val successState = FinalizationSucceededState
   override val failureState = WorkflowFinalizationFailedState
 
   override val successResponse = WorkflowFinalizationSucceededResponse
   override def failureResponse(reasons: Seq[Throwable]) = WorkflowFinalizationFailedResponse(reasons)
 
-  startWith(ReadyToFinalize, WorkflowFinalizationActorData(Map.empty, List.empty, Map.empty))
-
-  when(ReadyToFinalize) {
-    case Event(StartEngineFinalizationCommand, _) =>
-      // Create each FinalizationActor with the list of calls which ere assigned to that backend.
-      val backendFinalizationActors = backendWorkflowActors(backendAssignments)
-      backendFinalizationActors.keys foreach { _ ! Finalize }
-      goto(WorkflowFinalizing) using WorkflowFinalizationActorData(backendFinalizationActors, List.empty, Map.empty)
-    case Event(AbortEngineFinalizationCommand, _) =>
+  startWith(FinalizationPendingState, WorkflowFinalizationActorData(Map.empty, List.empty, Map.empty))
+
+  when(FinalizationPendingState) {
+    case Event(StartFinalizationCommand, _) =>
+      val backendFinalizationActors = backendWorkflowActors(workflowDescriptor.backendAssignments)
```

Comment:
Oh, oops - the variables were poorly named in WorkflowLifecycleActor due to copy/paste. But this method indirectly calls the `backendActor()` method which in `WorkflowFinalizationActor` creates `BackendWorkflowFinalizationActor`s.